src/avt/Database/Database/avtGenericDatabase.C

```diff
@@ -268,18 +273,20 @@ DebugDumpDatasetCollection(avtDatasetCollection &dsc, int ndoms,
     string dumpDir = avtDebugDumpOptions::GetDumpDirectory();
     if (dumpDir == "")
         dumpDir = ".";
-    oss << dumpDir << "/gdb." << std::setfill('0') << std::setw(4) << call_count << "." << phaseName << ".vtk";
-    string dumpFile = oss.str();
-    vtkDataSetWriter *dsw = vtkDataSetWriter::New();
+    vtkNew<vtkDataSetWriter> dsw;
     dsw->SetFileTypeToASCII();
-    dsw->SetFileName(dumpFile.c_str());
     for (int i = 0 ; i < ndoms; i++)
     {
-        vtkDataSet *ds = dsc.GetDataset(i, 0);
-        dsw->SetInputData(i, ds);
+        oss << dumpDir << "/gdb." << std::setfill('0') << std::setw(4)
```

Comment:
Can we just move the declaration of oss into the loop, here and then remove `oss.clear()` and `oss.str(std::string())` lines?